pkg/execution/state/redis_state/queue.go

```diff
@@ -3114,28 +3166,48 @@ func (q *queue) Scavenge(ctx context.Context, limit int) (int, error) {
 		// This is for backwards compatibility with the previous concurrency index item format
 		queueKey := partition
 		if !isKeyConcurrencyPointerItem(partition) {
-			queueKey = q.primaryQueueShard.RedisClient.kg.Concurrency("p", partition)
+			queueKey = kg.Concurrency("p", partition)
 		}
 
-		cmd := q.primaryQueueShard.RedisClient.unshardedRc.B().Zrange().
+		// Drop key queues from concurrency pointer - these should not be in here
+		if strings.HasPrefix(queueKey, "{q:v1}:concurrency:custom:") {
+			err := client.Do(ctx, client.B().Zrem().Key(kg.ConcurrencyIndex()).Member(partition).Build()).Error()
+			if err != nil {
+				resultErr = multierror.Append(resultErr, fmt.Errorf("error removing key queue '%s' from concurrency pointer: %w", partition, err))
+			}
+			continue
+		}
+
+		cmd := client.B().Zrange().
 			Key(queueKey).
 			Min("-inf").
 			Max(now).
 			Byscore().
-			Limit(0, 100).
+			Limit(0, ScavengeConcurrencyQueuePeekSize).
 			Build()
-		itemIDs, err := q.primaryQueueShard.RedisClient.unshardedRc.Do(ctx, cmd).AsStrSlice()
+		itemIDs, err := client.Do(ctx, cmd).AsStrSlice()
 		if err != nil && err != rueidis.Nil {
```

Comment:
so what happens here if it's an error but it's `rueidis.Nil`?
does it just move forward?

Reply:
If it's rueidis nil. the ZSET is empty and will be handled by the next clause